### DISCUSSION TITLE
Vanila style setState or getState in react

### DISCUSSION THREAD

**Discussion by StKostyantin** [pmndrs/zustand#2126] [Q&A]
Can I use a method from vanila js to set or get a store in a react component?
will it lead to any breakdowns or unexpected behavior?

For example, if I use setState in the handle function in react
```
const handleSome = async () => {
    useStore.setState({
      some: true;
    });
  };
```

or getState 
```

const handleSome = async () => {
   const { some } = useSomeStore.getState()
  };
```

**Comment by dai-shi** [comment]
They are supported & recommended.

1.

https://github.com/pmndrs/zustand/blob/602e1e9ebe7d006b6ab0c30951446df44315a517/docs/guides/practice-with-no-store-actions.md?plain=1#L2

2.

```js
  const handleSome = () => {
    const { some } = useSomeStore.getState()
    // ...
  };
```
Didn't find that one in docs. 😓

**Reply by StKostyantin** [reply]
I apologize, it was my mistake in the second example

**Reply by dai-shi** [reply]
No, I just meant I didn't find such recommendation in docs. Anyway, both are valid usages as you expected.
